rst/elf.rst

```diff
@@ -0,0 +1,409 @@
+.. contents::
+.. sectnum::
+
+====================================
+eBPF ELF Profile Specification, v0.1
+====================================
+
+The Executable and Linking Format (ELF) is specified in
+Tool Interface Standard (TIS), "Executable and Linking Format (ELF) Specification, Version 1.2", May 1995, https://refspecs.linuxbase.org/elf/elf.pdf.
+
+This document specifies version 0.1 of the eBPF profile for ELF files.
+
+Documentation conventions
+=========================
+
+All integer fields are unsigned.
+
+TEXT Sections
```

Comment:
Does this document need to talk about what it expects in the ELF header, or how the ELF header affects the .text sections?  For instance do we need the machine field to be a magic `EM_BPF` value (do we have one assigned)?  Or do we expect BPF programs to also exist in the same file as other 'native' programs?

How do we expect 32-/64-bit to change .text sections?  Even if they don't let us be obvious about it.

Similarly big/little endian.  How does that affect instruction format?  Data format?